src/ResourceManagement/Compute/Compute.Tests/ScenarioTests/OSProfileTests.cs

```diff
@@ -294,5 +320,71 @@ public void TestVMWithLinuxOSProfile()
                 }
             }
         }
+
+
+        public static string ReadFromEmbeddedResource(Type type, string resourceName)
+        {
+            string result;
+            using (Stream manifestResourceStream = type.Assembly.GetManifestResourceStream(type, resourceName) ?? type.Assembly.GetManifestResourceStream(resourceName))
+            {
+                using (StreamReader streamReader = new StreamReader(manifestResourceStream))
+                {
+                    result = streamReader.ReadToEnd();
+                }
+            }
+
+            return result;
+        }
+
+        private static string GetAutoLogonContent(uint logonCount, string userName, string password)
+        {
+            return string.Format(
+                "<AutoLogon>" +
+                "<Enabled>true</Enabled>" +
+                "<LogonCount>{0}</LogonCount>" +
+                "<Username>{1}</Username>" +
+                "<Password><Value>{2}</Value><PlainText>true</PlainText></Password>" +
+                "</AutoLogon>", logonCount, userName, password);
+        }
+    }
+
+    public static class SecretVaultHelper
+    {
+        // TODO: Please remove once KeyVault ARM client is ready and replace by actual calls to the service.
+        // Currently, for new recroding of TestVMWithWindowsOSProfile supporting WinRM with Https listeners, we need to
+        // have a certifcate in KeyVault. To do that, we:
+        // 1. Execute the script CreateKeyVaultAndCertificateScript.ps1   -SubscriptionName: <subscriptionName> 
+        //    using appropriate SubscriptionName  (you can change Region, ResourceGroupName, and KeyVaultName too)
+        // 2. Update static values below (e.g. VaultId and CertifcateUrl) with values returned when running the script
+        // 3. Complete the recording. 
+        // 4. After recording completes, please delete key vault you created by deleting the whole resource group.
+
+        public static string KeyVaultId = "/subscriptions/0e1dad42-1d4c-4329-9d38-ea44c03a519d/resourceGroups/pslibtestosprofile/providers/Microsoft.KeyVault/vaults/pslibtestkeyvault";
+
+        public static Uri CertificateUrl = new Uri("https://pslibtestkeyvault.vault.azure.net:443/secrets/WinRM/5c0b61f14941409aa752665210fd1319");
+
+#pragma warning disable 1998
```

Comment:
@yugangw-msft It's actually pretty common to disable warning 1998 in the build environment. Is it feasible for you to disable it at the SDK solution level? Thanks.